src/components/userlist/userlist.tsx

```diff
@@ -1,9 +1,55 @@
-import React from 'react';
+import React, { useEffect, useState, MouseEvent } from 'react';
+import "./userlist.css";
+
+interface User {
+    id: number,
+    first_name: string,
+    last_name: string,
+    email: string,
+    avatar: string
+}
 
 function Userlist() {
+
+const [users, setUsers] = useState<User[]>([]);
+const [page, setPage] = useState<Number>(1);
+
+const getUsers = async () => {
+    const res = await fetch(`https://reqres.in/api/users?page=${page}`);
+    const data = await res.json();
+    setUsers(data.data)
+};
+
+useEffect(() => {
+    getUsers();
+}, [page])
```

Comment:
Look here -  https://betterprogramming.pub/understanding-the-useeffect-dependency-array-2913da504c44